GardenApp/app/src/main/java/com/cs1530_group1/gardenapp/ViewSpeciesInfoActivity.java

```diff
@@ -53,54 +52,67 @@ protected void onCreate(Bundle savedInstanceState) {
             return;  //stop trying, nothing was received.
         }
 
+
+        updateTextViews();
+
+
+    }
+
+    /**
+     * this updates the TextViews displayed to the user with the correct information
+     *
+     */
+    protected void updateTextViews(){
+
+        //get the garden from App class
+        Garden garden = ((App) getApplication()).getGarden();
+
         //get the name TextView and append the species name
-        TextView name = (TextView) findViewById(R.id.species_name);
         String nameString = "<b>Name:</b><br />"+speciesName;
-        name.setText(Html.fromHtml(nameString)); //Html.fromHtml is needed to set bold strings
+        setText(R.id.species_name,nameString);
 
         //get the description TextView and append info
-        TextView descrpit = (TextView)findViewById(R.id.species_descript);
         String descriptString = "<b>Description:</b><br />" + garden.getDescription(speciesName);
-        descrpit.setText(Html.fromHtml(descriptString));
+        setText(R.id.species_descript,descriptString);
 
         //set type TextView
-        TextView type = (TextView)findViewById(R.id.species_type);
         String typeString = "<b>Type:</b><br />" + garden.getSpeciesType(speciesName);
-        type.setText(Html.fromHtml(typeString));
+        setText(R.id.species_type,typeString);
 
         //set sun TextView
-        TextView sun = (TextView)findViewById(R.id.species_sun);
         String sunString = "<b>Sun Level:</b><br />" + garden.getSunLevel(speciesName);
-        sun.setText(Html.fromHtml(sunString));
+        setText(R.id.species_sun,sunString);
 
         //set plant date TextView
-        TextView plantDate = (TextView)findViewById(R.id.species_plant_date);
         String plantDateString = "<b>Plant Date:</b><br />"+garden.getPlantDate(speciesName);
-        plantDate.setText(Html.fromHtml(plantDateString));
+        setText(R.id.species_plant_date,plantDateString);
 
         //set prune date TextView
-        TextView pruneDate = (TextView)findViewById(R.id.species_prune_date);
         String pruneDateString = "<b>Prune Date:</b><br />"+garden.getPruneDate(speciesName);
-        pruneDate.setText(Html.fromHtml(pruneDateString));
+        setText(R.id.species_prune_date,pruneDateString);
```

Comment:
So I've been doing some work into populating our plant database. After reviewing pruning, a date isn't really that important. What's important is a description of HOW to prune. I think we need to convert this into a String, and drop all the calendar events related to it. Pruning is often ongoing.